boards/posix/native_sim/doc/index.rst

```diff
@@ -379,6 +379,13 @@ The following peripherals are currently provided with this board:
 .. _net-tools:
    https://github.com/zephyrproject-rtos/net-tools
 
+**Offloaded sockets driver**
+  Offloaded sockets driver is an alternative to Ethernet driver. Instead of
```

Comment:
Yeah, perhaps the suggestion is nicer, although I can fully understand the original text too.

Reply:
Maybe more important would be to document/explain the pros and cons of each approach?

What would one be missing (if anything) when using this new approach over TUN? If there's hardly any limitation or "cons", maybe it would make sense to actually *recommend* people favour this new driver as it's much easier to setup?

Reply:
> One of the "cons" for this driver is that we avoid using the L2 layer, Ethernet in this case. This means that various parts of the stack are not tested / utilized, also using VLAN, ARP, gPTP, LLDP etc is not possible with this. So if only testing / using application protocols and not caring about lower level stuff, then this driver is nice. Big benefit is also that user does not need root access, so this makes running tests etc. in container much nicer.

exactly the kind of answer I was hoping to get... and that I would like to see captured in the docs somewhere! :) FWIW it can totally be a follow-up PR. It might work quite well to capture the pros/cons in table form, with something like ❌ / ✅ for the various items/features

Reply:
Applied @aescolar suggestion.

Reply:
Thanks.
For @kartben 's request, would it make sense to add something like:
``` 
  This driver is an alternative to the :ref:`TAP based ethernet driver
  <nsim_per_ethe>`. Instead of using a virtual network in the Linux side, this
  driver utilizes Linux's standard BSD socket API. With this, multiple Zephyr
  applications can communicate over the Linux loopback interface.
  The benefit of this approach is that root privileges are not required and
  that the process is connected to the same interface as other Linux processes
  instead of a virtual network, facilitating testing without the need for extra
  setup in the host. The drawback is that the L2 layer of Zephyr's networking
  stack is not exercised.
 ``` 
 ( just a proposal, either of you can have better and more accurate wording :) )

Reply:
Extended description with content above.